lib/app/features/feed/presentation/compose_tweet/compose_tweet_navigator.dart

```diff
@@ -1,37 +1,39 @@
 import 'dart:async';
 
+import 'package:flutter/widgets.dart';
 import 'package:flutter_modular/flutter_modular.dart';
 
 import '../../../mainboard/domain/states/mainboard_state.dart';
 import '../../../mainboard/domain/states/mainboard_store.dart';
+import '../../../mainboard/presentation/mainboard/mainboard_page.dart';
 
 abstract class ComposeTweetNavigator {
   factory ComposeTweetNavigator({
     required Uri? currentUri,
-    required MainboardStore mainboardStore,
   }) =>
       currentUri?.path.endsWith('/compose') == true
           ? _PageComposeTweetNavigator()
-          : _TabComposeTweetNavigator(mainboardStore: mainboardStore);
+          : _TabComposeTweetNavigator();
 
-  FutureOr<void> navigateToFeed();
+  FutureOr<void> navigateToFeed([BuildContext? context]);
 }
 
 class _PageComposeTweetNavigator implements ComposeTweetNavigator {
   @override
-  void navigateToFeed() {
+  void navigateToFeed([BuildContext? context]) {
     Modular.to.pop();
   }
 }
 
 class _TabComposeTweetNavigator implements ComposeTweetNavigator {
-  _TabComposeTweetNavigator({
-    required MainboardStore mainboardStore,
-  }) : _mainboardStore = mainboardStore;
-
-  final MainboardStore _mainboardStore;
   @override
-  Future<void> navigateToFeed() async {
-    await _mainboardStore.changePage(to: const MainboardState.feed());
+  Future<void> navigateToFeed([BuildContext? context]) async {
+    assert(context != null, 'context must be provided');
```

Comment:
O que esse assert vai fazer aqui? Ele não vai "quebrar" o código?

Por exemplo, se o contexto for `null`. O assert vai ser falso. Ele vai chegar no If da linha seguinte?

Reply:
Os `asserts` são úteis somente em tempo de desenvolvimento, no código produtivo todos são removidos pelo compilador.